src/components/scenes/CryptoExchangeScene.tsx

```diff
@@ -285,12 +285,14 @@ export class CryptoExchangeComponent extends React.Component<Props, State> {
         allowKeysOnlyMode={whichWallet === 'from'}
         filterActivation
       />
-    )).then(({ walletId, currencyCode }: WalletListResult) => {
-      if (walletId != null && currencyCode != null) {
-        return this.props.onSelectWallet(walletId, currencyCode, whichWallet)
-      }
-    })
-    return null
+    ))
+      .then(async result => {
+        const { walletId, currencyCode } = result
+        if (walletId != null && currencyCode != null) {
+          await this.props.onSelectWallet(walletId, currencyCode, whichWallet)
+        }
+      })
+      .catch(error => showError(error))
```

Comment:
Will this showError give us the stack to show it was called from CryptoExchangeScene? If not, this would be a good opportunity to create and use the onionError() function

Reply:
Yes, "CryptoExchangeScene.tsx:295" will show up in the stack, since we are doing `.catch(error => showError(error))`, instead of `.catch(showError)`.

We don't need any weird utility functions to handle these basic cases.